src/thor/trippathbuilder.cc

```diff
@@ -1047,10 +1045,13 @@ TripPathBuilder::Build(const AttributesController& controller,
                         (edge_shape.begin() + ((edge_begin_info.exists || is_first_edge) ? 0 : 1)),
                         edge_shape.end());
 
-      // If edge_begin_info.exists and not the first edge
-      // then increment begin_index
-      // since the previous end shape index should not equal
-      // the current begin shape index because of discontinuity
+      // Adjust the length of the trimmed edge
+      trip_edge->set_length(edge_length * kKmPerMeter *
```

Comment:
i wonder if this fixes a bug or creates a new one in `trace_attributes`. basically it trims the length property of the edge, but maybe in trace_attribues we want the actual edge length where as for regular routes we want the trimmed length. seems like its debatable anyway...

also does the trimming code also make sure there arent any duplicate shape points? i would expect if they both get trimmed to the same point it might duplicate that point in the final shape?